app/src/main/java/com/eveningoutpost/dexdrip/localeTasker/receiver/FireReceiver.java

```diff
@@ -162,8 +167,59 @@ public void onReceive(final Context context, final Intent intent) {
                             break;
 
                         case "SNOOZE":
-                            AlertPlayer.defaultSnooze();
-                            JoH.static_toast_long("SNOOZE from Tasker");
+                            // default: snoozes the length of the current alert, or the default alert length
+                            int minutes = -1;
+                            if (message_array.length > 1) {
+                                minutes = Integer.valueOf(message_array[1]);
+                                JoH.static_toast_long("SNOOZE from Tasker for "+minutes+" min");
+                            } else {
+                                JoH.static_toast_long("SNOOZE from Tasker");
+                            }
+                            AlertPlayer.getPlayer().Snooze(xdrip.getAppContext(), minutes);
+                            break;
+
+                        case "SNOOZE_LOW":
```

Comment:
Here, an enum would also be appropriate.

Reply:
Would you mind clarifying what you think should be an enum here? An enum representing all of the supported top-level messages inside tasker?

Reply:
> Would you mind clarifying what you think should be an enum here? An enum representing all of the supported top-level messages inside tasker?

Yes, that would be the correct way to implement. But keep it as is, for now.